uint32 state
string state_name
